src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1011,10 +1011,23 @@ else if (fieldClass == URI.class) {
             result = "URI";
         }
         else if (fieldClass == Pattern.class) {
-            result = "Regular Expression";
+            if ("SuppressionSingleFilter:checks".equals(checkProperty)
+                || "SuppressionXpathSingleFilter:files".equals(checkProperty)
+                || "IllegalTokenText:format".equals(checkProperty)) {
+                result = "Regular Expression";
+            }
+            else {
+                result = "Pattern";
+            }
         }
         else if (fieldClass == Pattern[].class) {
-            result = "Regular Expressions";
+            if ("ImportOrder:groups".equals(checkProperty)
+                || "ImportOrder:staticGroups".equals(checkProperty)) {
```

Comment:
why this things are exceptional?

Reply:
while discussion is not finished on how to deal with this Checks, lets keep them as they were on  "Regular Expression" (not real java type), See details in telegram group.

Reply:
as we already have such type "Regular Expressions" in doc, we will keep it and even place in metadata for now, to let plugins know that type is very special and specified in words only in chekcstyle doc.